sklearn/feature_extraction/hashing.py

```diff
@@ -148,6 +152,6 @@ def transform(self, raw_X, y=None):
         X = sp.csr_matrix((values, indices, indptr), dtype=self.dtype,
                           shape=(n_samples, self.n_features))
         X.sum_duplicates()  # also sorts the indices
-        if self.non_negative:
+        if self.non_negative is True:  # if non_negative == 'total', X>0 anyway
```

Comment:
Not necessarily true. We need to decide how to handle that case.

One solution is to add a further parameter... But honestly, if there is negative input, `non_negative` should be `False`.